xwiki-platform-core/xwiki-platform-index/xwiki-platform-index-api/src/main/java/org/xwiki/index/internal/DefaultTasksManager.java

```diff
@@ -135,7 +135,7 @@
         try {
             try {
                 this.tasksStore.get().addTask(wikiId, xWikiTask);
-            } catch (XWikiException e) {
+            } catch (Exception e) {
```

Comment:
This single change should be enough to make Oracle pass (since the task will be added in database). The rest of the PR aims at preventing empty string for the version column by appending a `-` before the version, and removing it afterward.